block/gs/adapter.go

```diff
@@ -173,6 +173,25 @@ func (a *Adapter) Remove(obj block.ObjectPointer) error {
 	return nil
 }
 
+func (a *Adapter) Copy(sourceObj, destinationObj block.ObjectPointer) error {
+	var err error
+	defer reportMetrics("Copy", time.Now(), nil, &err)
+	qualifiedDestinationKey, err := resolveNamespace(destinationObj)
+	if err != nil {
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("resolve destination: %w", err)
```
and similar below gives context on what name failed.